src/utils/use-multiselect.ts

```diff
@@ -0,0 +1,162 @@
+
+            const lastSelectedIdIndex = lastSelectedElementId
+                ? elementIds.indexOf(lastSelectedElementId)
+                : -1;
+            const clickedElementIdIndex = elementIds.indexOf(clickedElementId);
+
+            // if no lastSelectedId is found (first click, or unknown id), we only toggle clicked element
+            if (lastSelectedIdIndex < 0) {
+                toggleSelection(clickedElementId);
+                return;
+            }
+
+            // list of elements between lastClickedElement and clickedElement, both included
+            const elementsToToggle = elementIds.slice(
+                Math.min(lastSelectedIdIndex, clickedElementIdIndex),
+                Math.max(lastSelectedIdIndex, clickedElementIdIndex) + 1
+            );
+
+            if (selectedIds.has(clickedElementId)) {
+                // if clicked element is checked, we unchecked all elements between last clicked element and clicked element
+                unselectElements(elementsToToggle);
+            } else {
+                // if clicked element is unchecked, we check all elements between last clicked element and clicked element
+                selectElements(elementsToToggle);
+            }
+            setLastSelectedElementId(clickedElementId);
+        },
+        [
+            elementIds,
+            lastSelectedElementId,
+            selectedIds,
+            selectElements,
+            unselectElements,
+            toggleSelection,
+        ]
+    );
+
+    const handleShiftAndCtrlClick = (
+        clickEvent: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+        clickedElementId: string
+    ) => {
+        if (clickEvent.shiftKey) {
+            // if row is clicked while shift is pressed, range of rows selection is toggled, depending on clicked element state
+            handleShiftClick(clickedElementId);
+            // nothing else happens, hence the return
+            return;
+        }
+
+        if (clickEvent.ctrlKey) {
+            // if row is clicked while ctrl is pressed, row selection is toggled
+            // nothing else happens, hence the return right after
+            toggleSelection(clickedElementId);
+            return;
+        }
+    };
+
+    /**
+     * toggle selection for all element depending on current selectedIds
+     * @param elementsToSelectIds if defined, it will toggle only elementsToSelectIds instead of all elementIds
+     * @param forceSelectedIds if true, it will set selection to elementsToSelectIds/elementIds without checking current selectedIds
+     */
+    function toggleSelectAll(
+        elementsToSelectIds?: string[],
+        forceSelectedIds = false
+    ) {
+        if (selectedIds.size === 0 || forceSelectedIds) {
+            setSelectedIds(new Set(elementsToSelectIds ?? elementIds));
+        } else {
+            setSelectedIds(new Set());
+        }
+    }
+
+    return [
+        selectedIds,
+        toggleSelection,
```

Comment:
handleShiftAndCtrlClick is only to handle shift and ctrl clicks, it's not the same

Reply:
by the way I figured out we need it to manage selection when handling right click on a component... but it could be even easier if this hook manage rigth click event as well :P ... ok ok later...

Reply:
I think it's wrong, you're assuming we should always add an element to the selection if the element is right clicked
I don't think this is a common behaviour and it should not belong here